### DISCUSSION TITLE
General questions

### DISCUSSION THREAD

**Discussion by necroware** [necroware/gameport-adapter#55] [Q&A]
Ask your questions here.

**Comment by workingmanrob** [comment]
Hey Necroware

Very interested in getting my old sidewinder 3d pro plus hooked up via USB.

Are there any plans to sell a finished product or would I be better off buying the parts and putting it together myself?

Thanks for the project either way :beers:

**Reply by necroware** [reply]
Unfortunately, due to high license rates, it is quite complicated and expensive in Germany to produce and sell own hardware. Would I sell this parts for a reasonable price, then I would have to pay the fees out of my pocket. So, may be I'll find a way in the future, but currently you have to build it on your own. There are also many people around the globe who are making these devices and sell them on various platforms. If you search for may be you can find a seller in your region.

**Reply by workingmanrob** [reply]
I did find - https://oshpark.com/shared_projects/rzwJjMoP

and - https://www.tindie.com/products/retrofletch/pc-usb-gameport-adapter-sidewinder-retro-joystick/

but would of course prefer to support you if possible.

Found this as well - https://littlebigtech.net/posts/diy-gameport-to-usb-adapter/

**Reply by necroware** [reply]
Well, the only ways to support me currently is watching my YT channel or using Paypal or Patreon for donations. As I wrote, unfortunately I can't sell my adapters, nor any other of my projects legally from Germany. If I would, I have to pay fees, which are incredibly high and I would never compensate that costs by selling couple of adapters. Germany is complicated ;)

**Reply by Hedda** [reply]
> Are there any plans to sell a finished product or would I be better off buying the parts and putting it together myself?

Also see this related suggestion/request discussion here -> https://github.com/necroware/gameport-adapter/issues/46

> Unfortunately, due to high license rates, it is quite complicated and expensive in Germany to produce and sell own hardware.

As mentioned there, an option today for semi-hobby projects who already have a working prototype is to outsource the manufacturing and distribution to partners who offer one-stop-shop turnkey PCB and embedded circuit board manufacturers, and some even offers shipping directly from China or one of their distrubution points.

Some companies like for example PCBWay, JLCPCB, and Oshpark do offer such full shop services where they not only make and assembly your design but also act as a reseller with shipping directly to the customer via their own shop and store. See example:

https://www.pcbway.com/project/gifts-sell.html

https://www.pcbway.com/project/gifts.html

**Reply by necroware** [reply]
I'm currently not allowed by law to sell those adapters. To be able to do this, I have to pay very high fees in Germany, which I will never ever get back. Selling those parts would simply break project Necroware the neck financially and legaly. I know, that it would be nice, if I'd find a way, but unfortunately I have no time for that. So, please organise yourself in groups in your countries and build those adapters on your own. Meanwhile there are also sellers around the world, who sell ready to use parts based on my designs. Have fun.

**Reply by mamut-tme** [reply]
> To be able to do this, I have to pay very high fees in Germany, which I will never ever get back

Similar here in the neighboring Poland. If I want to sell a product I have to state that it conforms to all EU regulations and it has to have a CE certificate, which I can possibly give to myself without any testing. The big problem starts with Electromagnetic Compatibility (EMC). If someone makes a claim to our Communication Office (responsible mainly for radio stuff) that the device emits to much on radio bands and is possibly interferring with other devices I would be forced to take back all the sold products from the market, give the money back and pay for all fees. In theory it would be possible to make proper tests at a certified laboratory before starting selling, but this is a lot of money. And frankly speaking I doubt that the Arduino Micro is designed with having EMC in mind (especially the cheap clones). 
So really, oursourcing to companies like JLCPCB and then selling the products in the EU means shifting the responsibility for the product to ourselves. The only way out (which came to my mind several years ago) might be selling DIY kits, but even then the manufacturer (even repacker) is obliged to have CE certificates for all components (bye-bye connectors or resistors from aliexpress...).

Personally I have made my own version of necroware's adapter using a piece of a prototype PCB and some wire wrap wire. Maybe not so professional looking but working. I am thinking about designing a PCB, a bit differently than necroware is doing (I don't like the angled DSUB connectors, I much more prefer putting the pcb in between the pins of the connector), a colleague of mine also would like to have a copy so maybe I will order the PCBs somewhere. Anyone from Poland interested in such PCBs?

**Reply by Solderfinger** [reply]
First of all, thanks for the project and the work you put in it!
I'm not trying to pressure you but being a german myself, i can say that if you only make 10 or 20 of the boards and sell them on eBay, all you have to pay is eBay fees.
If you really don't want to get into the business, could you please upload a schematic so people can botch it together on veroboard for themselves?
Greetings and thanks again!

**Reply by MacGH23** [reply]
If you need one, please contact me ;-)
I have some left
see below ;-)

**Comment by Wile12** [comment]
Love this adapter, have the Sidewinder FFB Pro Stick and the Sidewinder FFB Wheel and finding
something for the joystick alone was almost impossible yet here is support for the wheel too!
Thank you for all your work on this project.
Tindie doesn't export to SA and PCBWAY seems to have the Micro-USB version, I'd like to try the USB-C so I
sent your project files to a local PCB fabricator here in South Africa but they say they only quote from gerbers which
I see your 2022-01-19 release last had. There seem to be 2 later releases since then and release notes did say no
hardware changes but I just want to confirm with you whether it would then be ok to just give them the gerbers
from your 2022-01-19 gameport-adapter-gerber-v0.2.1.zip file instead?
Will a pcb produced from that gerber be identical to one from your latest kicad from your 2023-03-26 release?
I have opened both kicad and gerber versions with viewers and didn't visually notice anything obviously different
but I'm completely inexperienced in pcb design so wouldn't know what to look out for.
Thanks again

**Reply by necroware** [reply]
Hi! There is no difference between Micro-USB and USB-C version, since the port is on the Arduino Pro Micro and not on the adapter PCB. So the type of USB port depends on the Pro Micro you use. The second number in the in the version 0.2.x means hardware version, the x is software version. So all 0.2.x releases are based on the same PCB, they differ only in firmware. You can find zipped gerber files in the releases:

https://github.com/necroware/gameport-adapter/releases

**Reply by Wile12** [reply]
Thank you Necroware.
I might have jumped the gun a bit posting before checking what PCBWAY actually offered.
I remember you mentioned the 3 Pro Micro versions (incl the wider mini-usb) in the 1st few minutes
of your 1st video and that being the sole determining factor of which type of connector you'll have.
Having never made use of PCBWAY (and having posted before creating an account with them) it seemed, from the 1 pic as though ordering from them might have included the Arduino. It was only after logging in and adding to cart that the next page showed that unless you ask for the assembly service, you get just the PCB's which is 100%.
Going to get those gerbers off to a few fabricators for quotes or I might now just go the PCBWAY route
(I mean... you seem to even get to choose your PCB in pink or even transparent!

**Comment by Hedda** [comment]
> Love this adapter, have the Sidewinder FFB Pro 

FYI, there is no ”force feedback” support yet, see -> https://github.com/necroware/gameport-adapter/issues/12

**Reply by Wile12** [reply]
Yes, thank you Hedda, I was aware, I just mentioned that those were the models I have.
Not having FFB is all but inconsequential to me at this stage since here's finally a solution that allows me to use my wheel again.
FFB working or not one day would just be gravy.

**Comment by MacGH23** [comment]
I just finished my design and have some left.
If anybody is interested send me a mail ;-)

**Comment by workingmanrob** [comment]
If anybody is in Montreal I have a bunch of spare blank PCBs I'd be willing
to give away.

Love my new necroware joystick mabob!

**Comment by pbaker3293** [comment]
does it have support for the interact V3 steering wheel?

and how do you put the software on it?

**Reply by fandjelo** [reply]
Hey, if something is not in the list of tested devices, than I don't have the device to test, nor did anyone else reported it as working.

**Comment by Solderfinger** [comment]
Hi MacGH23,
 
thanks for the kind offer.
I was actually thinking to integrate the Arduino and the four resistors directly into the Joystick and have a long USB cable to connect to my computer. So, i don't think i'll need a printed board, it would have just been nice to see where the resistors and the three bridges that act as microswitches go without "reverse engineering" the board.
 
 


Gesendet: Mittwoch, 19. Februar 2025 um 12:06
Von: MacGH23 ***@***.***>
An: necroware/gameport-adapter ***@***.***>
CC: Solderfinger ***@***.***>,Comment ***@***.***>
Betreff: Re: [necroware/gameport-adapter] General questions (Discussion #55)

 
If you need one, please contact me ;-)I have some leftsee below ;-)

**Reply by MacGH23** [reply]
Just install Kicad and open the project in the kicad folder.
there the schematics is shown

**Comment by JayErik** [comment]
Hello, 
first of all thanks for your effort and congratulations for this great project. I build my adapter based on your instructions and was going to use if for my Joystick Logitech Wingman Extreme. Unfortunately, only to find out that there are two different versions of this joystick in the field, one with and one without the "digital" word. The code supports the digital version, though, and i am not sure how to use the adapter for the non-digital version. Until now, I was also unsuccessful to find any technical specifications of both versions. 
Do you have some suggestions?

**Reply by fandjelo** [reply]
Hi. If it is not digital, it is probably analog. Try to use "Generic Analog" settings instead of Logitech/ADI

**Comment by Harbor-Gaming** [comment]
Hey, when do you believe this project will finally have FFB support? cuz i got a microsoft sidewinder force feedback wheel. BUT ITS THE GAMEPORT VERSION not the usb. it would be very important to me for this to have FFB support. thank you.

**Reply by necroware** [reply]
Unfortunately currently I have no spare time for this project, so it is on hold until better times. Sorry.

**Comment by MacGH23** [comment]
There is the old sidewinder FFB gameport adapter which support FFB.
I use it myself. 
You can ask if it is still available.
https://www.yard2usb.de/joomla/index.php/sidewinder-usb

**Reply by Harbor-Gaming** [reply]
If this ever gets ffb... will it require any hardware modifying? or it's just software?
And btw, the db15 conector of my wheel has some pins (for example pins 1 and 4) that simply arent conected to the arduino... is it meant to be like that?

**Reply by MacGH23** [reply]
About his one I don_'t know.
For my FFB I use the old adapter, for the 3dPro I use the necroware one.

**Comment by Shades2585** [comment]
Thanks for all your hard work. This looks fantastic!
I'm curious, is there any way the dip switch could be changes into a Dial of some type and a small led display could be on top showing what you have set it to? (sort of like the pie MT32)

Thanks everyone!

**Reply by MacGH23** [reply]
The easy one would be just LED connected to the switch. Show which switch is on.
Adding a LCD/OLED would need to add additional code only for the display.
The most people will set it only once.

**Reply by mamut-tme** [reply]
+changing the mode would require a re-detection of the USB device. If we change the mode, there is a completely different joystick detected (different amount of axes and buttons). I think that would require a restart of the microcontroller i.e. via watchdog or so.

**Comment by ataricom** [comment]
Does this support MIDI? I'm well aware of existing MIDI to USB cables, however those are tend to have problems with SysEx commands, at least with my older synths.

**Comment by TINYBOB1** [comment]
I recently got a Thrustmaster Millennium 3D Interceptor.  I know its not on the official list, but id be welling to test it and see if we can get it on the list.  Would any of the existing presets have a chance at working with it as a place to start or is this a lost cause?

**Comment by jetski23** [comment]
Hi,
I have got a Necroware-based adaptor from https://yard2usb.de/ - and I am trying to use my Sidewinder 3D-pro with I-War  (Indepenence War) - the original game which has the axes hard-coded. The problem that I have is that in the game the Z-axis is the throttle and the ZR axis is the rudder. Could I swap these by changing the values in the file sidewinder.h would that be in lines 312 to 316? if yes would changing 


```
// bit 35-36 + bit 40-46: z-axis (value 0-511)
    state.axes[2] = map(bits(35, 2) << 7 | bits(40, 7), 0, 511, 0, 1023);

 // bit 32-34 + bit 48-54: throttle-axis (value 0-1023)
    state.axes[3] = bits(32, 3) << 7 | bits(48, 7);
```

to 
  
```
// bit 35-36 + bit 40-46: z-axis (value 0-511)
    state.axes[3] = map(bits(35, 2) << 7 | bits(40, 7), 0, 511, 0, 1023);
   
// bit 32-34 + bit 48-54: throttle-axis (value 0-1023)
    state.axes[2] = bits(32, 3) << 7 | bits(48, 7);
```

work?

**Reply by necroware** [reply]
yes, that should work

**Reply by jetski23** [reply]
Thanks! I will try and let you know...

**Reply by jetski23** [reply]
Thanks, tried it and it worked...

**Comment by brecht-engels** [comment]
Hello @necroware
I have the Sidewinder Precision FFB Pro, but the DB15-pin connector is broken, so I cut it off and threw it away. A year later, I want to repair it and solder a new 15-pin connector to it, but I don't know which wire color goes to which pin. Could someone give me more information about this?

**Reply by Hedda** [reply]
@brecht-engels that is by deffinition not a general question about Necroware's GamePort Adapter but instead a very specific question about something only indirectly related, so you are going of on a tangent and at least really need to start a separate new discussion thread for it. Regardless I also believe that it would serve you to try posting that question on Reddit too if you want to get an answer more quickly.

**Comment by 444444432423** [comment]
Hi, ive been trying to get my sidewinder 3d pro plus to connect to my computer, but it is the model that doesnt have native usb support (the product id starts with 97462). is there any way to get this to work? I have an arduino micro pro (the 5v 16MHz one) and ive tried about 3 other projects but none have seemed to work even after poking and prodding for multiple days. any and all help is appreciated!!

**Reply by Hedda** [reply]
> ive been trying to get my sidewinder 3d pro plus to connect to my computer, but it is the model that doesnt have native usb support (the product id starts with 97462). is there any way to get this to work?

@444444432423 For reference, I do not think that you should ask when "Sidewinder 3D Pro Plus" is specifically listed in this project's readme file under the section that lists which joysticks does this adapter support -> https://github.com/necroware/gameport-adapter/blob/main/README.md

Readme also explicitly explain that this adapter in this project is built around Arduino Pro Micro, which uses the same ATmega32U4 microcontroller as Leonardo, so double check your Arduino Pro Micro is based on ATmega32U4.
